apps/communications/contacts/test/unit/views/details/boot_test.js

```diff
@@ -0,0 +1,203 @@
+    });
+
+    teardown(function() {
+      LazyLoader.load.restore();
+    });
+
+    test(' > LazyLoader must be called 2 times', function() {
+      // We must have 2 calls to LazyLoader
+      assert.isTrue(stub.calledTwice);
+    });
+
+    test(' > First call must ensure localization', function() {
+      // First of all we need to bring l10n library for localization
+      var firstCall = stub.getCall(0);
+      assert.isTrue(Array.isArray(firstCall.args[0]));
+      assert.equal(firstCall.args[0][0], '/shared/js/l10n.js');
+    });
+
+    test(' > Second call must load the template', function() {
+      // Secondly we lazy load the panel, so the param must be the element
+      // of the DOM to fill with the template
+      var secondCall = stub.getCall(1);
+      assert.equal(secondCall.args[0][0].id, 'view-contact-details');
+    });
+  });
+
+  suite('Loaded', function() {
+    var realDetailsController, realDetailsUI, realMozSetMessage;
+    suiteSetup(function() {
+      realDetailsController = window.Detail;
+      realDetailsUI = window.DetailsUI;
+
+      window.Details = {
+        init: function foo() {}
+      };
+
+      window.DetailsUI = {
+        init: function foo() {}
+      };
+
+      realMozSetMessage = navigator.mozSetMessageHandler;
+      navigator.mozSetMessageHandler = function foo() {};
+    });
+
+    suiteTeardown(function() {
+      window.Detail = realDetailsController;
+      window.DetailsUI = realDetailsUI;
+      navigator.mozSetMessageHandler = realMozSetMessage;
+
+      realDetailsController = null;
+      realDetailsUI = null;
+      realMozSetMessage = null;
+    });
+
+    setup(function() {
+      this.sinon.stub(LazyLoader, 'load', function(files, callback) {
+        callback();
+      });
+    });
+
+    test(' > Controller is initialized', function() {
+      var DetailsSpy = this.sinon.spy(Details, 'init');
+      window.dispatchEvent(new CustomEvent('load'));
+      assert.isTrue(DetailsSpy.calledOnce);
+    });
+
+    test(' > UI is initialized', function() {
+      var DetailsUISpy = this.sinon.spy(DetailsUI, 'init');
+      window.dispatchEvent(new CustomEvent('load'));
+      assert.isTrue(DetailsUISpy.calledOnce);
+    });
+  });
+
+  suite('Activity', function() {
+    var realDetailsController, realDetailsUI, realContactsService, testContact;
+    var fakeActivity = {
+      source: {
+        data: {
+          params: {
+            'id': 'testID'
+          }
+        }
+      }
+    };
+
+    suiteSetup(function() {
+      realDetailsController = window.Details;
+      realDetailsUI = window.DetailsUI;
+      realContactsService = window.ContactsService;
+      testContact = new mozContact();
+      testContact.givenName = ['test'];
+
+      window.Details = {
+        init: function init() {},
+        setActivity: function setActivity() {}
+      };
+
+      window.DetailsUI = {
+        init: function init() {},
+        render: function render() {}
```

Comment:
Why don't we initialized the whole Mockup in https://github.com/mozilla-b2g/gaia/pull/30750/files#diff-cd0522fc76b47adba110970b54bf9d4cR68 ?